components/post_view/post_message_view/index.js

```diff
@@ -28,6 +30,11 @@ function makeMapStateToProps() {
 
         const user = getCurrentUser(state);
 
+        let channelNamesMap = getChannelsNameMapInCurrentTeam(state);
+        if (ownProps.post.props && ownProps.post.props.channel_mentions) {
+            channelNamesMap = Object.assign({}, ownProps.post.props.channel_mentions, channelNamesMap);
```

Comment:
This should be cached by a selector. `mapStateToProps` is called on each store change, so it's creating new objects here constantly. It would be something like

```
const makeGetChannelsNameMap = createSelector(
    getChannelsNameMapInCurrentTeam,
    (state, channelMentionsHint) => channelMentionsHint
    (channelNamesMap, channelMentionsHint) => {
        ....
     }
);
```